libs/klv/src/main/java/org/codice/ddf/libs/klv/data/numerical/KlvIntegerEncodedFloatingPoint.java

```diff
@@ -64,11 +66,37 @@ public KlvIntegerEncodedFloatingPoint(final KlvNumericalDataElement<?> klvRawDat
         this.actualRangeMax = actualRangeMax;
     }
 
-    @Override
-    protected void decodeValue(final Klv klv) {
-        klvRawDataValue.decodeValue(klv);
-        value = convert(klvRawDataValue.getValue()
-                .longValue(), encodedRangeMin, encodedRangeMax, actualRangeMin, actualRangeMax);
+    /**
+     * Constructs a {@code KlvIntegerEncodedFloatingPoint} representing a KLV element containing a
+     * floating-point value that has been encoded as an integer value.
+     * <p>
+     * The value returned by this {@code KlvIntegerEncodedFloatingPoint} is calculated as follows:
+     * <li>The value returned by {@code klvRawDataValue} is converted to a <strong>long</strong></li>
+     * <li>Scale the value by calculating: (value - {@code encodedRangeMin}) / ({@code encodedRangeMax} - {@code encodedRangeMin})</li>
+     * <li>Apply the scaled value to the actual range: scaled value * ({@code actualRangeMax} - {@code actualRangeMin}) + {@code actualRangeMin}</li>
+     *
+     * @param klvRawDataValue the {@link KlvNumericalDataElement} that retrieves the raw numerical value
+     * @param encodedRangeMin the minimum integer-encoded value
+     * @param encodedRangeMax the maximum integer-encoded value
+     * @param actualRangeMin  the minimum decoded floating-point value
+     * @param actualRangeMax  the maximum decoded floating-point value
+     * @param errorIndicatorValue value that indicates an encoded error
+     */
+    public KlvIntegerEncodedFloatingPoint(final KlvNumericalDataElement<?> klvRawDataValue,
+            final long encodedRangeMin, final long encodedRangeMax, final double actualRangeMin,
+            final double actualRangeMax, Optional<Double> errorIndicatorValue) {
+        super(klvRawDataValue.getKey(), klvRawDataValue.getName(), errorIndicatorValue);
```

Comment:
The other constructor could be simplified to `this(klvRawDataValue, encodedRangeMin, encodedRangeMax, actualRangeMin, actualRangeMax, Optional.empty());`